crates/via-router/src/visitor.rs

```diff
@@ -3,7 +3,32 @@ use crate::{
     routes::{Node, RouteStore},
 };
 
+/// Inserts a match into the `Vec` of matches for a `Visitor` and then
+/// continues to match descendants of `$node` that matched the path segment.
+///
+/// This macro is used to prevent duplication between match branches of nodes
+/// that have a `Pattern::Dynamic` or `Pattern::Static` pattern.
+macro_rules! insert_match_and_continue {
```

Comment:
I don't think duplication is a bad thing, especially since we're calling `match_at_depth` recursively. However, I do believe that preventing drift between the match branches of `Pattern::Dynamic` and `Pattern::Static` is worth the cognitive overhead of a macro. As always, I'm open to suggestions from the community if there is a better way to achieve what I'm trying to achieve here.